composer.json

```diff
@@ -57,7 +57,8 @@
         "markbaker/complex": "^1.4",
         "markbaker/matrix": "^1.2",
         "psr/simple-cache": "^1.0",
-        "guzzlehttp/guzzle": "^7.0"
+        "psr/http-client": "^1.0",
+        "guzzlehttp/psr7": "^1.6"
```

Comment:
`guzzlehttp/psr7` is safe, compared to `guzzlehttp/guzzle`, because it only contains the tools to create the `Request` object, which would be cumbersome if we had to create it directly in PhpSpreadsheet.

Reply:
If we do the PSR way, we hat to do it all the way. That means dropping all references to Guzzle and instead rely on https://www.php-fig.org/psr/psr-17/ to create Request object.

Reply:
Even though I definitely agree on PSR-17, there are some kind of maintainability issues with it: this means that PhpSpreadsheet would have to have its own factory, plus its own Request class, plus its own Response class for testing.

Here, relying on `guzzlehttp/psr7` is safer than the entire Guzzle stack, because `guzzlehttp/psr7` depends only on `ralouphie/getallheaders` version 2 **and** 3, making it compatible with almost all packages that might already have it in their dependency tree. For instance, Drupal needs it since 8.8, Slim needs it too, but they depend on its v3. There are very few packages that need its v2, and yet guzzle's PSR-7 implementation is still compatible with it.

Seems really fair in terms of balance between maintainability and dependency tree 😉